src/main/java/fi/vrk/xroad/monitor/parser/SharedParamsParser.java

```diff
@@ -0,0 +1,71 @@
+package fi.vrk.xroad.monitor.parser;
+
+import lombok.extern.slf4j.Slf4j;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class for parsing X-Road shared-params.xml
+ */
+@Slf4j
+public class SharedParamsParser {
+
+  private final String filename;
+
+  public SharedParamsParser(String filename){
+    this.filename = filename;
+  }
+
+  public List<SecurityServerInfo> parse() throws ParserConfigurationException, IOException, SAXException {
```

Comment:
Would have been nice to have a brief summaryish explanation about what information is parsed, using what logic.